### DISCUSSION TITLE
HOW TO CHANGE THE OPTIMIZER IN YOLOV8 OTHER THAN SGD

### DISCUSSION THREAD

**Discussion by vijayakrishna162003** [ultralytics/ultralytics#2733] [Q&A]
CAN ANYONE TELL ME HOW TO CHANGE THE OPTIMIZER TO ADAM OR RMSProp OTHER THAN SGD ?

**Comment by HichTala** [comment]
The easiest way is to add it as an argument in the train method:

```
model.train(data="coco128.yaml", epochs=3, optimizer='Adam')
```

You can choose between ```SGD```, ```Adam```, ```AdamW```, ```RMSProp``` for the optimizer argument.

You can refer to the [documentation](https://docs.ultralytics.com/modes/train/#arguments) to see all the configurable hyperparameters,

Hope this help !

**Reply by glenn-jocher** [reply]
@HichTala hello! To change the optimizer in YOLOv8 to Adam or RMSprop, you can adjust the training configuration file (usually a `.yaml` file). In this file, you'll find a section for the optimizer settings where you can specify the type of optimizer you want to use.

For example, to use the Adam optimizer, you would change the optimizer setting to `adam`. Similarly, for RMSprop, you would set it to `rmsprop`. Make sure to adjust any optimizer-specific hyperparameters as needed for your training.

Remember to review the documentation on the Train mode for more details on configuring your training runs. Happy training! 😊🚀
